src/fluree/db/util/core.cljc

```diff
@@ -211,9 +211,9 @@
   "Keywordizes string contexts so they merge correctly with other keyword
   contexts."
   [context-type context]
-  (if (= :string context-type)
-    (keywordize-keys context)
-    context))
+  (if (= :keyword context-type)
+    context
+    (keywordize-keys context)))
```

Comment:
I'm sort of confused about this, `keywordize-keys` already checks if keys are a string. Is this just meant to save that work when it's known to be unnecessary?

And the breakage here was because we were skipping out on keywordization when we shouldn't have?

Sort of feels weird to me to have an implicit enum of context types (keyword or string), but we just sometimes violate that by having no context type (or do they sometimes end up mixed?), and then we have two runtime-check-style fns to clean things up.

(Not a blocker for merging this, more just thinking out loud/trying to make sure I understand.)

Reply:
Yeah we could just always call `keywordize-keys` but if `:context-type :keyword` is set (the default) then it _should_ be unnecessary.

The case where we have no context-type is when we're reifying from storage. So it's a side effect of these functions be used in two rather different scenarios:

1. A user is creating a ledger for their own use and supplies a context-type or gets the default of `:keyword`.
2. We are internally reloading an existing ledger from storage. We have full control over the context in this case, so we could make it always strings or always keywords and set the context-type explicitly.

I just seem to keep playing mixed-keyword-and-string-keys whack-a-mole in here. So trying to balance that against not doing more work than is necessary. Definitely an area for additional cleanup / refactoring.